doc/advanced-configuration/listen.md

```diff
@@ -0,0 +1,738 @@
+* **Default:** `true`
+* **Example:** `tls.disconnect_on_failure = false`
+
+Requires setting `tls.verify_mode`.  When set to `false`, it allows the client to connect even though the certificate verification failed. It is then up to the authentication layer to accept or reject the client connection. This behaviour mimcs the FastTLS one.
+
+#### `listen.c2s.tls.certfile`
+* **Syntax:** string, path in the file system
+* **Default:** not set
+* **Example:** `tls.certfile = "server.pem"`
+
+Path to the X509 PEM file with a certificate and a private key (not protected by a password). If the certificate is signed by an intermediate CA, you should specify here the whole CA chain by concatenating all public keys together and appending the private key after that.
+
+#### `listen.c2s.tls.cacertfile`
+* **Syntax:** string, path in the file system
+* **Default:** not set
+* **Example:** `tls.cacertfile = "ca.pem"`
+
+Path to the X509 PEM file with a CA chain that will be used to verify clients. Won't have any effect if `verify_peer` is not enabled.
+
+#### `listen.c2s.tls.dhfile`
+* **Syntax:** string, path in the file system
+* **Default:** not set
+* **Example:** `tls.dhfile = "dh.pem"`
+
+Path to the Diffie Hellman parameter file.
+
+#### `listen.c2s.tls.ciphers` - for `fast_tls`
+* **Syntax:** string with the OpenSSL cipher suite specification
+* **Default:** `"TLSv1.2:TLSv1.3"`
+* **Example:** `tls.ciphers = "ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384"`
+
+Cipher suites to use with StartTLS or TLS. Please refer to the [OpenSSL documentation](http://www.openssl.org/docs/man1.0.2/apps/ciphers.html) for the cipher string format.
+
+#### `listen.c2s.tls.ciphers` - for `just_tls`
+* **Syntax:** array of tables with the following keys: `cipher`, `key_exchange`, `mac`, `prf` and string values.
+* **Default:** not set, all supported cipher suites are accepted
+* **Example:** `tls.ciphers = "[{cipher = "aes_25_gcm", key_exchange = "any", max = "aead", "prf = sha384"}]"`
+
+Cipher suites to use with StartTLS or TLS. For allowed values, see the [Erlang/OTP SSL documentation](https://erlang.org/doc/man/ssl.html#type-ciphers)
+
+#### `listen.c2s.tls.versions` - for `just_tls`
+* **Syntax:** array of strings
+* **Default:** not set, all supported versions are accepted
+* **Example:** `tls.versions = ["tlsv1.2", "tlsv1.3"]`
+
+TLS versions to use with StartTLS or TLS. For allowed values, see the [Erlang/OTP SSL documentation](https://erlang.org/doc/man/ssl.html#type-protocol_version)
+
+#### `listen.c2s.tls.crl_files` - for `just_tls`
+* **Syntax:** array of strings, paths in the file system
+* **Default:** not set
+* **Example:** `tls.crl_files = ["certs.crl"]`
+
+Specifies the paths to Certificate Revocation Lists.
+
+#### `listen.c2s.tls.protocol_options` - for `fast_tls`
+* **Syntax:** array of strings
+* **Default:** `["no_sslv2", "no_sslv3", "no_tlsv1", "no_tlsv1_1"]`
+* **Example:** `tls.protocol_options = ["no_tlsv1", "no_tlsv1_1"]`
```

Comment:
the `just_tls` and `fast_tls` options are interweaven here with each other - maybe it could be better to group them by library. Right now the `cipher` option makes sense but overall we have a structure like:
```
fast_tls_1
just_tls_1
just_tls_2
just_tls_3
fast_tls_2
```
And I think we could make it a bit more organized

Reply:
Honestly, I juggled them over and over again, trying to find the best order. I could reorder them once more... I just wanted to follow the order in example docs, the parser, the Erlang docs etc...